src/python/pants/backend/python/dependency_inference/parse_python_imports.py

```diff
@@ -68,8 +72,10 @@ async def parse_python_imports(request: ParsePythonImportsRequest) -> ParsedPyth
     )
     # See above for where we explicitly encoded as utf8. Even though utf8 is the
     # default for decode(), we make that explicit here for emphasis.
-    process_output = process_result.stdout.decode("utf8")
-    return ParsedPythonImports(json.loads(process_output) if process_output else {})
+    process_output = process_result.stdout.decode("utf8") or "{}"
+    return ParsedPythonImports(
+        {imp: ParsedPythonImportInfo(**info) for imp, info in json.loads(process_output).items()}
```

Comment:
You can avoid creating an intermediate `dict` by using a generator expression

```suggestion
        (imp, ParsedPythonImportInfo(**info)) for imp, info in json.loads(process_output).items()
```